hooks/useEstimateGasCost.ts

```diff
@@ -41,39 +41,34 @@ export default function useEstimateGasCost() {
 	);
 
 	const estimateEthersContractTxCost = useCallback(
-		async (contract: Contract, method: string, params: any[]): Promise<Wei> => {
+		async (
+			contract: Contract,
+			method: string,
+			params: any[],
+			buffer?: number
+		): Promise<[Wei, Wei | undefined]> => {
 			try {
 				if (!contract?.estimateGas[method]) throw new Error('Invalid contract method');
 				const gasLimit = await contract?.estimateGas[method](...params);
 				const metaTx = await contract?.populateTransaction[method](...params);
-				if (!metaTx || !gasLimit || !gasPrice?.gasPrice) return zeroBN;
+				if (!metaTx || !gasLimit || !gasPrice?.gasPrice) return [zeroBN, undefined];
+				const gasLimitWithBuffer = buffer ? gasLimit.add(buffer) : gasLimit;
 				const l1Fee = await sdk.transactions.getOptimismLayerOneFees({
 					...metaTx,
 					gasPrice: gasPrice?.gasPrice?.toNumber(),
-					gasLimit: Number(gasLimit),
+					gasLimit: Number(gasLimitWithBuffer),
 				});
-				return getTransactionPrice(gasPrice, gasLimit, ethPriceRate, l1Fee) || zeroBN;
+				return [
+					getTransactionPrice(gasPrice, gasLimit, ethPriceRate, l1Fee) || zeroBN,
+					wei(gasLimitWithBuffer, 0, true),
+				];
 			} catch (err) {
 				logError(err);
```

Comment:
I think this might be my code originally but what do you think about just letting this error out rather than returning zero and undefined?

Reply:
Done, updated to an object